docs/source/extension/extension_migration.rst

```diff
@@ -48,6 +48,9 @@ bumped their major version (following semver convention). We want to point out p
    of ``ICompletableAttributes`` interface by ``ICompletionProvider``. To create a completer provider
    for JupyterLab, users need to implement the interface ``ICompletionProvider`` and then register
    this provider with ``ICompletionProviderManager`` token.
+- ``@jupyterlab/codeeditor`` from 3.x to 4.x
+   * Remove ``ISelectionStyle`` (and therefore ``defaultSelectionStyle`` and ``IEditor.selectionStyle``). This was envisaged
+     for real-time collaboration. But this is not used in the final implementation.
```

Comment:
Are those being replaced by an equivalent functionality provided in upstream packages? More generally, what is the status of RTC support for multi-selection (https://github.com/jupyterlab/jupyterlab/issues/10239)?

Reply:
And selection in general? Do we have an issue for tracking that selection highlights are not implemented in RTC and therefore collaborators cannot see what others selected?

Reply:
Each RTC library comes with its own way to exchange non-content information. Yjs is exchanging information about user and cursor position through the awareness. And the transfer of information between CodeMirror editor and Yjs is done through an external library `y-codemirror.next`; for example the line that set the primary cursor in the awareness is:

https://github.com/yjs/y-codemirror.next/blob/318bbf7b684ea64d015f8f40f68e6820461e1d52/src/y-remote-selections.js#L170

The binding for CodeMirror 6 takes the form of additional CodeMirror 6 extensions.

To share selection and multi-cursors, we need to add that information in the awareness.

---

Regarding styling, as the user color is a thing now, it makes no sense to be able to a color for the user selection. That selection color should be the same (probably adding some transparency).

> I'm working on a follow-up of CodeMirror 6 migration to remove the usage of global object in favor of our plugin system. And then I plan to have a PR to improve the collaborative user display (improving the user name display, adding selection, and multi-selections).